Add automation standards for PowerShell scripts (#224)
Azure/Azure-Proactive-Resiliency-Library: docs/content/contributing/_index.md

If you need support with validating a query, please reach out to the APRL team via the [APRL GitHub General Question/Feedback Form](https://github.com/Azure/Azure-Proactive-Resiliency-Library/issues/new?assignees=&labels=feedback%2C+question&projects=&template=general-question-feedback----.md&title=%E2%9D%93%F0%9F%91%82+Question%2FFeedback+-+PLEASE+CHANGE+ME+TO+SOMETHING+DESCRIPTIVE)
{{< /alert >}}

### Azure PowerShell Scripts

1. All PowerShell scripts should have two comments at the top of the script, one comment stating `Azure PowerShell script` and another comment providing a description of the script results returned. For example:

    ```powershell
    // Azure PowerShell script
    // Provides a list of Azure Container Registry resources that do not have soft delete enabled
    ```

1. Scripts should only return resources that do not adhere to the APRL recommendation. For example, if the recommendation is to enable soft delete for Azure Container Registries, the associated scripts should only return Azure Container Registry resources that do not have soft delete enabled.

> 1. Scripts should only include codes to return resources that do not adhere to the APRL recommendation. Scripts should not include the support code such as sign-in to Azure ([Connect-AzAccount](https://learn.microsoft.com/en-us/powershell/module/az.accounts/connect-azaccount), Login-AzAccount), subscription selection ([Set-AzContext](https://learn.microsoft.com/en-us/powershell/module/az.accounts/set-azcontext), Select-AzSubscription), etc. Those codes should be executed outside of the APRL recommendation PowerShell script.

Reviewer: ```suggestion
1. Scripts should exclusively contain code to retrieve resources that do not comply with the APRL recommendation. They should not include supporting code, such as Azure sign-in ([Connect-AzAccount](https://learn.microsoft.com/en-us/powershell/module/az.accounts/connect-azaccount), Login-AzAccount) or subscription selection ([Set-AzContext](https://learn.microsoft.com/en-us/powershell/module/az.accounts/set-azcontext), Select-AzSubscription). Execute these codes separately from the APRL recommendation PowerShell script.

> 1. The script should returned the result as an array of `PSCustomObject` data type, and each result object should only include the following properties:

Reviewer: ```suggestion
1. The script should return the result as an array of the `PSCustomObject` data type, with each result object containing only the following properties:
```
